core/src/main/scala/org/apache/spark/scheduler/TaskDescription.scala

```diff
@@ -52,7 +55,43 @@ private[spark] class TaskDescription(
     val addedFiles: Map[String, Long],
     val addedJars: Map[String, Long],
     val properties: Properties,
-    val serializedTask: ByteBuffer) {
+    private var serializedTask_ : ByteBuffer) extends  Logging {
```

Comment:
While there is value in making the task a member variable of this class (to reduce code changes on the master), I think that making the serializedTask a member variable adds too much confusion -- because the conditions under which it is set are a bit difficult to reason about.  How about removing it as a member variable, and then encode() can do the serialization of the Task itself, and decode can return a (TaskDescription, ByteBuffer) pair, where the 2nd thing is the serialized task (and then that can be passed on its own into executor.launchTask)?  @squito thoughts?  I'd find that more clear because then no one needs to reason about when serializedTask_ is and is not set to something.